components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -104,6 +107,11 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
             // addingRoles = newRoles AND allExistingRoles
             Collection<String> addingRoles = getRolesToAdd(userStoreManager, newRoles);
 
+            String idp = attributes.get(FrameworkConstants.IDP_ID);
+            String subjectVal = attributes.get(FrameworkConstants.ASSOCIATED_ID);
```

Comment:
Can't we directly remove here ? Doesn't remove return the respective object after removing ?